src/java.base/share/classes/java/lang/Long.java

```diff
@@ -446,33 +488,105 @@ private static void formatUnsignedLong0UTF16(long val, int shift, byte[] buf, in
     }
 
     static String fastUUID(long lsb, long msb) {
+        char[] hex256 = HEX256;
+
+        char i0 = hex256[((int) (msb >> 56)) & 0xff];
+        char i1 = hex256[((int) (msb >> 48)) & 0xff];
+        char i2 = hex256[((int) (msb >> 40)) & 0xff];
+        char i3 = hex256[((int) (msb >> 32)) & 0xff];
+        char i4 = hex256[(((int) msb) >> 24) & 0xff];
+        char i5 = hex256[(((int) msb) >> 16) & 0xff];
+        char i6 = hex256[(((int) msb) >> 8) & 0xff];
+        char i7 = hex256[((int) msb) & 0xff];
+        char i8 = hex256[(((int) (lsb >> 56))) & 0xff];
+        char i9 = hex256[(((int) (lsb >> 48))) & 0xff];
+        char i10 = hex256[(((int) (lsb >> 40))) & 0xff];
+        char i11 = hex256[((int) (lsb >> 32)) & 0xff];
+        char i12 = hex256[(((int) lsb) >> 24) & 0xff];
+        char i13 = hex256[(((int) lsb) >> 16) & 0xff];
+        char i14 = hex256[(((int) lsb) >> 8) & 0xff];
+        char i15 = hex256[((int) lsb) & 0xff];
+
         if (COMPACT_STRINGS) {
             byte[] buf = new byte[36];
-            formatUnsignedLong0(lsb,        4, buf, 24, 12);
-            formatUnsignedLong0(lsb >>> 48, 4, buf, 19, 4);
-            formatUnsignedLong0(msb,        4, buf, 14, 4);
-            formatUnsignedLong0(msb >>> 16, 4, buf, 9,  4);
-            formatUnsignedLong0(msb >>> 32, 4, buf, 0,  8);
 
-            buf[23] = '-';
-            buf[18] = '-';
+            buf[0] = (byte) (i0 >> 8);
+            buf[1] = (byte) i0;
+            buf[2] = (byte) (i1 >> 8);
+            buf[3] = (byte) i1;
+            buf[4] = (byte) (i2 >> 8);
+            buf[5] = (byte) i2;
+            buf[6] = (byte) (i3 >> 8);
+            buf[7] = (byte) i3;
+            buf[8] = '-';
+            buf[9] = (byte) (i4 >> 8);
+            buf[10] = (byte) i4;
+            buf[11] = (byte) (i5 >> 8);
+            buf[12] = (byte) i5;
             buf[13] = '-';
-            buf[8]  = '-';
+            buf[14] = (byte) (i6 >> 8);
+            buf[15] = (byte) i6;
+            buf[16] = (byte) (i7 >> 8);
+            buf[17] = (byte) i7;
+            buf[18] = '-';
+            buf[19] = (byte) (i8 >> 8);
+            buf[20] = (byte) i8;
+            buf[21] = (byte) (i9 >> 8);
+            buf[22] = (byte) i9;
+            buf[23] = '-';
+            buf[24] = (byte) (i10 >> 8);
+            buf[25] = (byte) i10;
+            buf[26] = (byte) (i11 >> 8);
+            buf[27] = (byte) i11;
+            buf[28] = (byte) (i12 >> 8);
+            buf[29] = (byte) i12;
+            buf[30] = (byte) (i13 >> 8);
+            buf[31] = (byte) i13;
+            buf[32] = (byte) (i14 >> 8);
+            buf[33] = (byte) i14;
+            buf[34] = (byte) (i15 >> 8);
+            buf[35] = (byte) i15;
```

Comment:
You may be able to use `jdk.internal.util.ByteArray` to simplify code and improve performance:

```suggestion
            ByteArray.setChar(buf, 0, i0);
            ByteArray.setChar(buf, 2, i1);
            ByteArray.setChar(buf, 4, i2);
            ByteArray.setChar(buf, 6, i3);
            buf[8] = '-';
            ByteArray.setChar(buf, 9, i4);
            ByteArray.setChar(buf, 11, i5);
            buf[13] = '-';
            ByteArray.setChar(buf, 14, i6);
            ByteArray.setChar(buf, 16, i7);
            buf[18] = '-';
            ByteArray.setChar(buf, 19, i8);
            ByteArray.setChar(buf, 21, i9);
            buf[23] = '-';
            ByteArray.setChar(buf, 24, i10);
            ByteArray.setChar(buf, 26, i11);
            ByteArray.setChar(buf, 28, i12);
            ByteArray.setChar(buf, 30, i13);
            ByteArray.setChar(buf, 32, i14);
            ByteArray.setChar(buf, 34, i15);
```

I measured the throughput of `UUID::toString()` using JMH on my linux server (CPU: R7-5800X):

```diff
  Benchmark                 Mode  Cnt      Score     Error   Units
- UUIDUtilsBenchmark.test  thrpt    5  76117.237 ± 461.679  ops/ms
+ UUIDUtilsBenchmark.test  thrpt    5  87291.496 ± 729.527  ops/ms
```

The result of this benchmark test is that using `ByteArray` can improve peak throughput by nearly 15%.

Note: `ByteArray` uses `VarHandle` internally, and the impact on startup time is unknown, so more testing is needed.